connectivity/FEATURE_BLE/source/cordio/source/GattServerImpl.cpp

```diff
@@ -687,6 +731,8 @@ ble_error_t GattServer::read(
         return BLE_ERROR_NONE;
     }
 
+    tr_info("result=%s", mbed_trace_array(buffer, *buffer_length));
```

Comment:
you are prining to early, the result has not been found yet,
at the same time there is a return on line 731 that doesn't print the value